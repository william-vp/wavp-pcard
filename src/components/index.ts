import {ProductCard as ProductCardComponent} from "./ProductCard";
import {ProductTitle} from "./ProductTitle";
import {ProductImage} from "./ProductImage";
import {ProductButtons} from "./ProductButtons";
import {ProductCardMainProps} from "../interfaces/interfaces";

export {ProductTitle} from "./ProductTitle";
export {ProductImage} from "./ProductImage";
export {ProductButtons} from "./ProductButtons";

export const ProductCard: ProductCardMainProps = Object.assign(ProductCardComponent, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
});
export default ProductCard;