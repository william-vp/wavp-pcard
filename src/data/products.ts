import {Product} from "../interfaces/interfaces";
import coffeMug from "../coffee-mug.png";
import coffeMug2 from "../coffee-mug2.png";

const product = {
    id: '1',
    title: 'Coffe Mug',
    img: coffeMug
}

const product2 = {
    id: '2',
    title: 'Coffe Mug 2',
    img: coffeMug2
}
export const products: Product[] = [product, product2]