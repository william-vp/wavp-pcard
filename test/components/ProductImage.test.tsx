import React from 'react';
import {ProductImage, ProductCard} from "../../src";
import render from "react-test-renderer";
import { product2} from "../data/products";

describe('ProductImage', () => {
    test('mostrar componente', () => {
        const wrapper = render.create(
            <ProductImage img={'ruta'} />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('mostrar componente con imagen de producto', () => {
        const wrapper = render.create(
            <ProductCard product={product2} >
                { () => (
                    <ProductImage/>
                )}
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
