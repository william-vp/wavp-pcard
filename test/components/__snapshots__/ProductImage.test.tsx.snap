// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage mostrar componente 1`] = `
<img
  alt="image"
  className="productImg "
  src="ruta"
/>
`;

exports[`ProductImage mostrar componente con imagen de producto 1`] = `
<div
  className="productCard "
>
  <img
    alt="Coffe Mug 2"
    className="productImg "
    src="coffeMug2"
  />
</div>
`;
